app/scripts/services/ownerReferences.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("OwnerReferencesService", function() {
+    var getOwnerReferences = function(apiObject) {
+      return _.get(apiObject, 'metadata.ownerReferences');
+    };
+    return {
+      getOwnerReferences: getOwnerReferences,
+      groupByControllerUID: function(apiObjects) {
+        return _.groupBy(apiObjects, function(apiObject) {
+          var ownerRefs = getOwnerReferences(apiObject);
+          var controllerRef = _.find(ownerRefs, { controller: true });
```

Comment:
is it possible (in theory) for more than one controller to claim ownership, if that were the case you would want the item to appear under each group and not just the first one you find.   Would be curious if the API prevents you from having more than one ownerRef with controller true.  Could try manually adding an ownerRef on a pod that points to a random controller that exists.